DataFormats/METReco/src/HcalNoiseHPD.cc

```diff
@@ -110,28 +110,28 @@ const edm::RefVector<HBHERecHitCollection> HcalNoiseHPD::recHits(void) const
   return rechits_;
 }
   
-float HcalNoiseHPD::recHitEnergy(float threshold) const
+float HcalNoiseHPD::recHitEnergy(const float threshold) const
 {
-  float total=0.0;
+  double total=0.0;
   for(edm::RefVector<HBHERecHitCollection>::const_iterator it=rechits_.begin(); it!=rechits_.end(); ++it) {
-    float energy=(*it)->energy();
+    const float energy=(*it)->eraw();
     if(energy>=threshold) total+=energy;
   }
   return total;
 }
 
-float HcalNoiseHPD::recHitEnergyFailR45(float threshold) const
+float HcalNoiseHPD::recHitEnergyFailR45(const float threshold) const
```

Comment:
[I don't recall asking for this change, so, just my 2c notes on this]

const float& doesn't make much sense, it's less efficient than what it is now.
declaring something that's not supposed to be changed as a const is a good practice (in this case for the scope of the function, not for the users of the function).